### PR TITLE
[GLUTEN-3743][VL] Support pushed bool filter in scan

### PR DESCRIPTION
**Pull request by Yohahaha**
## What changes were proposed in this pull request?

(Fixes: \#3743)

## How was this patch tested?

Add a test.

### CODE REVIEW

**Comment by Yohahaha** [comment]
Run Gluten Clickhouse CI

File: gluten-core/src/main/scala/io/glutenproject/execution/BasicScanExecTransformer.scala [hunk 1]
```diff
@@ -113,6 +114,11 @@ trait BasicScanExecTransformer extends LeafTransformSupport with BaseDataSource
     }.asJava
     // Will put all filter expressions into an AND expression
     val transformer = filterExprs()
+      .map {
+        case ar: AttributeReference if ar.dataType == BooleanType =>
+          EqualNullSafe(ar, Literal.TrueLiteral)
```
Comment (rui-mo) [inline gluten-core/src/main/scala/io/glutenproject/execution/BasicScanExecTransformer.scala:119]: Could you help double-check if null or false is returned for null input?

**Review by rui-mo — COMMENTED** [review]
Thanks. Just one question, and please resolve the conflicts.

File: gluten-core/src/main/scala/io/glutenproject/execution/BasicScanExecTransformer.scala (hunk 1 above)
Comment (Yohahaha) [reply gluten-core/src/main/scala/io/glutenproject/execution/BasicScanExecTransformer.scala:119]: add ut to cover this case.

**Review by rui-mo — APPROVED** [review]

File: gluten-core/src/main/scala/io/glutenproject/execution/BasicScanExecTransformer.scala [hunk 2]
```diff
@@ -113,6 +114,11 @@ trait BasicScanExecTransformer extends LeafTransformSupport with BaseDataSource
     }.asJava
     // Will put all filter expressions into an AND expression
     val transformer = filterExprs()
+      .map {
+        case ar: AttributeReference if ar.dataType == BooleanType =>
```
Comment (ulysses-you) [inline gluten-core/src/main/scala/io/glutenproject/execution/BasicScanExecTransformer.scala:118]: does it work with not ? e.g., `WHERE !b`

Comment (Yohahaha) [reply gluten-core/src/main/scala/io/glutenproject/execution/BasicScanExecTransformer.scala:118]: yes, `where !b` equal to `where b=null`

**Pull request closed without merging**